.github/workflows/preview.yml

```diff
@@ -0,0 +1,35 @@
+name: Deploy to Preview
+
+on:
+  push:
+    branches: [main]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [build]
```

Comment:
There's no `build` job in this file (or any other workflow file - build.yml has a job called `test`) - unfortunately I don't think GitHub Actions currently has a way to include a job from another file so you'll need to include the job in this file. On Welsh Petitions the GHA file for building pull requests uses this to restrict it to just PRs that target the main branch:

```
on:
  pull_request:
    branches: [main]
```

Obviously you'll need to change `main` to `master` if you want to use a similar config in build.yml.

Reply:
Given we have the tests run on all commits and PRs (which can't be merged into master unless they are passing), I thought I'd speed up deploys by not running the test suite again (and duplicate all the config for running those tests).

Missed this `needs` though, so I've removed it. Sound reasonable?

Reply:
That's fine but merging a PR counts as a push to the master branch so it will run tests anyway - is that what you intended?

Reply:
I don't mind the tests running, I just don't think we need the deploy task to depend on the tests passing (as they will already have done on the branch)